File: ruby/3_methods_practice_problems.rb
# Use built-in methods for strings and numbers

# 1. Write a program that asks the user to enter a word, then prints that word with all capital letters.

print "Enter a word: "
word = gets.chomp
p word.upcase

# 2. Write a program that asks the user to enter a number, then prints "That's a big number" if the number is greater than 100.

print "Enter a number: "
num = gets.chomp

if num.to_i > 100
  p "That's a big number"
end

# 3. Write a program that asks the user to enter two numbers, then prints the numbers added together.

print "Enter a number: "
n1 = gets.chomp
print "Enter a second number: "
n2 = gets.chomp
p n1.to_i + n2.to_i

# 4. Write a program that asks the user to enter a word, then prints that word in reverse order.

print "Enter a word: "
word = gets.chomp
p word.reverse

# 5. Write a program that asks the user to enter a number, then prints the number times 10.

print "Enter a number: "
num = gets.chomp
p num.to_i * 10

# 6. Write a program that asks the user to enter two words, then prints both words on the same line in all capital letters.

print "Enter a word: "
w1 = gets.chomp
print "Enter another word: "
w2 = gets.chomp
p w1.upcase + " " + w2.upcase

# 7. Write a program that asks the user to enter a word, then prints the number of letters in the word.

print "Enter a word: "
word = gets.chomp
p word.length

# 8. Write a program that asks the user to enter a number, then prints "That's a negative number" if the number is less than 0.

print "Enter a number: "
num = gets.chomp
if num.to_i < 0
  p "That's a negative number"
end

# 9. Write a program that asks the user to enter two numbers, then prints the two numbers multiplied together.

print "Enter a number: "
n1 = gets.chomp
print "Enter another number: "
n2 = gets.chomp
p n1.to_i * n2.to_i

# 10. Write a program that asks the user to enter a word, then prints "That's a long word" if the word has more than 5 letters.

print "Enter a word: "
word = gets.chomp
if word.length > 5
  p "That's a long word"
end

# SOLUTIONS: https://gist.github.com/peterxjang/1539a3ad79728ba4fb68dd8d07279c29

File: javascript/3_methods_practice_problems.js
// Use built-in methods for strings and numbers

// 1. Write a program that asks the user to enter a word, then prints that word with all capital letters.\

// 2. Write a program that asks the user to enter a number, then prints "That's a big number" if the number is greater than 100.

// 3. Write a program that asks the user to enter two numbers, then prints the numbers added together.

// 4. Write a program that asks the user to enter a word, then prints that word in reverse order.

// 5. Write a program that asks the user to enter a number, then prints the number times 10.

// 6. Write a program that asks the user to enter two words, then prints both words on the same line in all capital letters.

// 7. Write a program that asks the user to enter a word, then prints the number of letters in the word.

// 8. Write a program that asks the user to enter a number, then prints "That's a negative number" if the number is less than 0.

// 9. Write a program that asks the user to enter two numbers, then prints the two numbers multiplied together.

// 10. Write a program that asks the user to enter a word, then prints "That's a long word" if the word has more than 5 letters.

// JavaScript solutions: https://gist.github.com/peterxjang/d30d686d6dbe194dbb89465849b6926f

// NOTES FOR BROWSER vs. NODE EXECUTION
// // IF RUNNING IN NODE:
// (Learn boiler-plate solution, utilizing callback methods.)
// const readline = require("readline").createInterface({
//   input: process.stdin,
//   output: process.stdout,
// });

// readline.question("Who are you?", (name) => {
//   console.log(`Hey there ${name}!`);
//   readline.close();
// });

// ~ or ~

// // IF RUNNING IN NODE WITH "prompt-sync" LIBRARY
// const prompt = require("prompt-sync")();
// const word = prompt("Enter a word: ");
// console.log(word.toUpperCase());

// // IF RUNNING IN BROWSER:
// var num1 = window.prompt("Enter a number: ");
// var num2 = window.prompt("Enter another number: ");
// console.log(parseInt(num1) + parseInt(num2));
